src/Symfony/Component/Security/Core/User/PasswordAuthenticatedUserInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\User;
+
+/**
+ * For users that can be authenticated using a password.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+interface PasswordAuthenticatedUserInterface
+{
+    /**
+     * Returns the hashed password used to authenticate the user.
+     *
+     * Usually on authentication, a plain-text password will be compared to this value.
+     *
+     * @return string|null The hashed password or null (if not set or erased)
+     */
+    public function getPassword(): ?string;
```

Comment:
Maybe we should take this opportunity to rename the method to `getPasswordHash` to make it clear that this is not the cleartext password ?

Reply:
That would mean removing `getPassword()` from the `UserInterface` in 5.3, which I think would be a too much impacting BC break. Not sure it's worth it